3-0-spring-framework/3-2-1-account-rest-api/src/test/java/com/bobocode/AccountRestControllerTest.java

```diff
@@ -0,0 +1,164 @@
+    @Autowired
+    private WebApplicationContext applicationContext;
+
+    @Autowired
+    private InMemoryAccountDao accountDao;
+
+    private MockMvc mockMvc;
+
+    @BeforeEach
+    void setup() {
+        mockMvc = MockMvcBuilders.webAppContextSetup(applicationContext).build();
+        accountDao.clear();
+    }
+
+    @Test
+    @Order(1)
+    @DisplayName("AccountRestController is marked as @RestController")
+    void accountRestControllerAnnotation() {
+        RestController restController = AccountRestController.class.getAnnotation(RestController.class);
+
+        assertNotNull(restController);
+    }
+
+    @Test
+    @Order(2)
+    @DisplayName("AccountRestController is annotated and mapped with @RequestMapping")
+    void accountRestControllerRequestMapping() {
+        RequestMapping requestMapping = AccountRestController.class.getAnnotation(RequestMapping.class);
+
+        assertNotNull(requestMapping);
+        assertThat(requestMapping.value().length).isEqualTo(1);
+        assertThat(requestMapping.value()).contains("/accounts");
+    }
+
+    @Test
+    @Order(3)
+    @DisplayName("AccountDao is injected using constructor")
+    void accountDaoInjection() {
+        Constructor<?> constructor = AccountRestController.class.getConstructors()[0];
+
+        assertThat(constructor.getParameterTypes()).contains(AccountDao.class);
+    }
+
+    @Test
+    @Order(4)
+    @DisplayName("Getting all accounts is implemented")
+    void getAllAccounts() throws Exception {
+        Account account1 = create("Johnny", "Boy", "jboy@gmail.com");
+        Account account2 = create("Okko", "Bay", "obay@gmail.com");
+        accountDao.save(account1);
+        accountDao.save(account2);
+
+        mockMvc.perform(get("/accounts"))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.[*].email").value(hasItems("jboy@gmail.com", "obay@gmail.com")));
+    }
+
+    @Test
+    @Order(5)
+    @DisplayName("Getting all accounts response status is OK")
+    void getAccountsResponseStatusCode() throws Exception {
+        mockMvc.perform(get("/accounts").accept(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk());
+    }
+
+    @Test
+    @Order(6)
+    @DisplayName("Getting account by Id with path variable is implemented")
+    void getById() throws Exception {
+        Account account = create("Johnny", "Boy", "jboy@gmail.com");
+        accountDao.save(account);
+
+        mockMvc.perform(get(String.format("/accounts/%d", account.getId())))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.id").value(account.getId()))
+                .andExpect(jsonPath("$.email").value("jboy@gmail.com"))
+                .andExpect(jsonPath("$.firstName").value("Johnny"))
+                .andExpect(jsonPath("$.lastName").value("Boy"));
+    }
+
+    @Test
+    @Order(7)
+    @DisplayName("Creating account returns corresponding HTTP status")
```

Comment:
@mathodcoast `Creating account returns the corresponding HTTP status – 201`

Reply:
let's use `create account` instead of `creating account` or `update account` vs `updating account`